src/js/view/controls/components/volumetooltipicon.ts

```diff
@@ -81,12 +90,10 @@ export default class VolumeTooltipIcon extends TooltipIcon {
         this.addSliderHandlers(this.horizontalSlider.uiOver);
         this.addSliderHandlers(this.verticalSlider.uiOver);
 
-        this.onAudioMode(null, _model.get('audioMode'));
-
-        this._model.on('change:audioMode', this.onAudioMode, this);
+        this._model.on('change:audioMode', this.setHorizontalSliderTabIndex, this);
     }
 
-    onAudioMode(model: Model | null, val: boolean): void {
+    setHorizontalSliderTabIndex(model: Model | null, val: boolean): void {
         const tabIndex = val ? 0 : -1;
         setAttribute(this.horizontalContainer, 'tabindex', tabIndex);
```

Comment:
If `change:audioMode` is trigged with `false` while `_model.get('horizontalVolumeSlider')` is true,  the index should still be set.

```
        this._model.change('audioMode', this.updateSlider, this);
    }

    updateSlider(model: Model, audioMode: boolean): void {
	const hasHorizontalSlider = _model.get('horizontalVolumeSlider') || audioMode;
        const tabIndex = val ? 0 : -1;
         if (hasHorizontalSlider) {
            this.horizontalContainer.style.display = 'flex';
            this.tooltip.style.display = 'none';
        } else {
           // do the opposite
        }
        setAttribute(this.horizontalContainer, 'tabindex', tabIndex);
        // does this tab index need to be dynamic if the element is hidden?
        // set this.tooltip tab index too?

Reply:
Nice call!

It also looks like we don't need the `tabindex` to be dynamic so I removed that. We already set it to 0 by default when the sliders are created above. We do something similar with the buttons. Their tab indexes are all set 0 when created, but they can't be tabbed to when they are hidden.